packages/components/src/toolbar/toolbar-item/types.ts

```diff
@@ -0,0 +1,26 @@
+/**
+ * External dependencies
+ */
+import type {
+	ReactElement,
+	ReactNode,
+	ElementType,
+	HTMLAttributes,
+	RefAttributes,
+} from 'react';
+
+export interface ToolbarItemProps
+	extends Omit< HTMLAttributes< any >, 'children' > {
+	/**
+	 * Component type that will be used to render the toolbar item.
+	 */
+	as?: ElementType;
```

Comment:
I think we've typed the `as` prop like this before:

```suggestion
	as?: ElementType | keyof JSX.IntrinsicElements;
```

Reply:
`ElementType` already includes `keyof JSX.IntrinsicElements`.